clippy_lints/src/matches.rs

```diff
@@ -223,6 +223,26 @@ declare_clippy_lint! {
     "a wildcard enum match arm using `_`"
 }
 
+declare_clippy_lint! {
+    /// **What it does:** Checks for wildcard pattern used with others patterns in same match arm.
+    ///
+    /// **Why is this bad?** Wildcard pattern already covers any other pattern as it will match anyway.
+    /// It makes the code less readable, especially to spot wildcard pattern use in match arm.
+    ///
+    /// **Known problems:** None.
+    ///
+    /// **Example:**
+    /// ```rust
+    /// match "foo" {
+    ///     "a" => {},
+    ///     "bar" | _ => {},
+    /// }
+    /// ```
+    pub PATS_WITH_WILD_MATCH_ARM,
+    restriction,
```

Comment:
This needs to be in the `complexity` group since the suggestion simplifies the code.

Reply:
Ok. Kind of hard at first in which group we should add lints, but that's why big brothers/sisters are here ;) thanks!